program/src/token.rs

```diff
@@ -257,4 +269,36 @@ pub mod test {
         // Invalid character.
         assert!(Lamports::from_str("lol, sol").is_err());
     }
+
+    #[test]
+    fn test_rational_cmp() {
+        // Construct x and y such that x < y.
+        let x = Rational {
+            numerator: 1 << 53,
+            denominator: 1,
+        };
+        let y = Rational {
+            numerator: x.numerator + 1,
+            denominator: x.denominator,
+        };
+        assert_eq!(x.partial_cmp(&y), Some(std::cmp::Ordering::Less));
+    }
+
+    #[test]
+    fn test_division_by_zero_cmp() {
+        let x = Rational {
+            numerator: 1,
+            denominator: 0,
+        };
+        let y = Rational {
+            numerator: x.numerator,
+            denominator: x.denominator + 1,
+        };
+        assert_eq!(x.partial_cmp(&y), None);
```

Comment:
Please also test the reverse case of `y.partial_cmp(&x)`, that’s one of those places where a subtle copy-paste bug might hide. In fact, let’s also add the reverse case in the `test_rational_cmp` test above.